maxminddb/extension/maxminddb.c

```diff
@@ -209,6 +211,28 @@ static PyObject *Reader_close(PyObject *self, PyObject *UNUSED(args))
         mmdb_obj->mmdb = NULL;
     }
 
+    mmdb_obj->closed = Py_True;
+
+    Py_RETURN_NONE;
+}
+
+static PyObject *Reader__enter__(PyObject *self, PyObject *UNUSED(args))
+{
+    Reader_obj *mmdb_obj = (Reader_obj *)self;
+
+    if(mmdb_obj->closed == Py_True) {
+        PyErr_SetString(PyExc_ValueError,
+                        "Attempt to reopen a closed MaxMind DB.");
+        return NULL;
+    }
+
+    Py_INCREF(self);
```

Comment:
I wasn't sure either. I got a segfault when not using Py_INCREF though. 

Looking at other peoples code (pysqlite) it looks like they do the same thing:
https://github.com/ghaering/pysqlite/blob/e728ffbcaeb7bfae1d6b7165369bd0ae16cabf95/src/connection.c#1554